tests/core/test_connections.py

```diff
@@ -83,31 +137,56 @@ class Test(object):
     @classmethod
     def setup_class(cls):
         Connections.deregister_all()
-        Connections.register("dummy", DummyConnectionPlugin)
-        Connections.register("dummy2", DummyConnectionPlugin)
-        Connections.register("dummy_no_overrides", DummyConnectionPlugin)
+        Connections.register(DummyConnectionPlugin)
+        Connections.register(DummyConnectionPlugin, "dummy2")
+        Connections.register(DummyConnectionPlugin, "dummy_no_overrides")
 
     def test_open_and_close_connection(self, nornir):
         nr = nornir.filter(name="dev2.group_1")
         r = nr.run(task=open_and_close_connection, num_workers=1)
         assert len(r) == 1
         assert not r.failed
 
+    def test_open_and_close_connection_same_plugin_different_name(self, nornir):
+        nr = nornir.filter(name="dev2.group_1")
+        r = nr.run(task=open_and_close_connection_same_plugin_different_name)
+        assert len(r) == 1
+        assert not r.failed
+
     def test_open_connection_twice(self, nornir):
         nr = nornir.filter(name="dev2.group_1")
         r = nr.run(task=open_connection_twice, num_workers=1)
         assert len(r) == 1
         assert not r.failed
 
+    def test_auto_two_connections_dummy_plugin(self, nornir):
+        with nornir.filter(name="dev2.group_1") as nr:
+            r1 = nr.run(task=dummy_task)
+            assert r1["dev2.group_1"].result == "ok"
+            r2 = nr.run(task=dummy_task, conn_name="secondary_dummy")
```

Comment:
We need to find a better way. An alternative would be to have a task_configuration parameter and a TaskConfiguration object where we allow tweaking the behavior.